### ISSUE TITLE
Dreambooth isn't working (google colab)

### ISSUE THREAD

**Issue by mdlieber99** [TheLastBen/fast-stable-diffusion#2951]
both gemini and ChatGPT think there is an issue with the det.py script.  It fails in the "model download" section whether I give it a path or a link.  given the other recent comments it looks like this might be a problem on a1111 as well.  Here is the error when I give it the path:

Detecting model version...
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
[<ipython-input-3-11e838c2e30a>](https://8fx7iev2ixe-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab_20241111-060148_RC00_695270385#) in <cell line: 144>()
    218     wget.download('https://github.com/TheLastBen/fast-stable-diffusion/raw/main/Dreambooth/det.py')
    219     print('Detecting model version...')
--> 220     Custom_Model_Version=check_output('python det.py '+sftnsr+' --MODEL_PATH '+str(MODEL_PATH), shell=True).decode('utf-8').replace('\n', '')
    221     clear_output()
    222     print(''+Custom_Model_Version+' Detected')

1 frames
[/usr/lib/python3.10/subprocess.py](https://8fx7iev2ixe-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab_20241111-060148_RC00_695270385#) in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    524         retcode = process.poll()
    525         if check and retcode:
--> 526             raise CalledProcessError(retcode, process.args,
    527                                      output=stdout, stderr=stderr)
    528     return CompletedProcess(process.args, retcode, stdout, stderr)

CalledProcessError: Command 'python det.py --from_safetensors --MODEL_PATH /content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/cyberrealistic_v50.safetensors' returned non-zero exit status 1.

**Comment by vadar007** [comment]
Yup, that fixed it for me (Issue #2950)! How did you pin done the need to install this additional piece of software?

**Comment by mdlieber99** [comment]
> Yup, that fixed it for me (Issue #2950)! How did you pin down the need to install this additional piece/version of software?

So chatGPT said it was about the det.py script.  I noticed that others were raising issues on this page about a1111 not working and that was also about the det.py script.  Someone offered this solution on that page so I tried it here and it worked.

**Comment by unco-ctr** [comment]
> ok - I added !pip install protobuf==3.20.3 before the model download step and now it seems to be working

Could you please help me putting the line, where exactly did you add it? I am getting the same error and adding it doesn't help. Thanks!

**Comment by vadar007** [comment]
I inserted as the first line in the colab Dependencies code:

**!pip install protobuf==3.20.3**
from IPython.utils import capture
import time
import os

You still get some dependencies warning messages but it runs....

**Comment by unco-ctr** [comment]
> I inserted as the first line in the colab Dependencies code:
> 
> **!pip install protobuf==3.20.3** from IPython.utils import capture import time import os
> 
> You still get some dependencies warning messages but it runs....

That is amazing, it works! Perhaps I was doing something wrong, but adding these to the beginning of the Model Download step didn't help. Thank you!

**Comment by mdlieber99** [comment]
Is dreambooth still working for you guys?  It broke again yesterday for me but no one else has brought up an issue

**Comment by vadar007** [comment]
Yep, been using it constantly today. What is the error message(s) you are getting? Mind you, I am running off a copy of the colab so if the source colab itself was modified, I probably won't see it.

**Comment by mdlieber99** [comment]
Thanks, Vadar.  Are you using protobuf 3.20.3 still or did you downgrade to
3.19.0?



Matthew D. Lieberman, Ph.D. (twitter <http://twitter.com/social_brains>)
Director, UCLA Social Cognitive Neuroscience Laboratory
<http://www.uclascnlab.com>
Co-Founder, Resonance Inc. <https://www.resonance.info>
Editor-in-Chief, Social Cognitive and Affective Neuroscience
<https://academic.oup.com/scan>
Zoom - https://ucla.zoom.us/my/lieberman

"Being skeptical means demanding evidence, not ignoring it"  -Bobby Azarian

"I would rather have questions that can't be answered than answers that
can't be questioned" -Richard Feynman

"When I went to school, they asked me what I wanted to be when I grew up. I
wrote down ‘happy.’ They told me I didn’t understand the assignment, and I
told them they didn’t understand life."  -John Lennon

**Comment by vadar007** [comment]
Sticking with protobuf 3.20.3
